IWXXM/sigmet.xsd

```diff
@@ -1,497 +1,407 @@
+			</xs:extension>
+		</xs:complexContent>
+	</xs:complexType>
+	<xs:complexType name="SIGMETPropertyType">
+		<xs:sequence minOccurs="0">
+			<xs:element ref="iwxxm:SIGMET"/>
+		</xs:sequence>
+		<xs:attributeGroup ref="gml:AssociationAttributeGroup"/>
+		<xs:attributeGroup ref="gml:OwnershipAttributeGroup"/>
+	</xs:complexType>
+	<xs:element name="TropicalCycloneSIGMET" type="iwxxm:TropicalCycloneSIGMETType" substitutionGroup="iwxxm:SIGMET">
+		<xs:annotation>
+			<xs:documentation>A SIGMET that reports the presence of a tropical cyclone conditions.  This extends the base SIGMET type by including additional information necessary for tropical cyclones.</xs:documentation>
+		</xs:annotation>
+	</xs:element>
+	<xs:complexType name="TropicalCycloneSIGMETType">
+		<xs:complexContent>
+			<xs:extension base="iwxxm:SIGMETType">
+				<xs:sequence>
+					<xs:element name="supplementaryAnalysisCollection" minOccurs="0" maxOccurs="1">
+						<xs:annotation>
+							<xs:documentation>Observed and/or forecast conditions of the cumulonimbus clouds associated with a tropical cyclone.  To be used for more than one cumulonimbus clouds simultaneously affecting the FIR concerned.</xs:documentation>
+						</xs:annotation>
+						<xs:complexType>
+							<xs:complexContent>
+								<xs:extension base="gml:AbstractMemberType">
+									<xs:sequence>
+										<xs:element ref="iwxxm:analysisAndForecastPositionAnalysis"/>
+									</xs:sequence>
+								</xs:extension>
+							</xs:complexContent>
+						</xs:complexType>
+					</xs:element>
+					<xs:element name="tropicalCyclone" type="metce:TropicalCyclonePropertyType" minOccurs="0" maxOccurs="1">
+						<xs:annotation>
+							<xs:documentation>The tropical cyclone being reported in this SIGMET, required in all cases other than cancellations. When reporting the tropical cyclone name, 'NN' may be used if the tropical cyclone is unnamed</xs:documentation>
+						</xs:annotation>
+					</xs:element>
+				</xs:sequence>
+			</xs:extension>
+		</xs:complexContent>
+	</xs:complexType>
+	<xs:complexType name="TropicalCycloneSIGMETPropertyType">
+		<xs:sequence minOccurs="0">
+			<xs:element ref="iwxxm:TropicalCycloneSIGMET"/>
+		</xs:sequence>
+		<xs:attributeGroup ref="gml:AssociationAttributeGroup"/>
+		<xs:attributeGroup ref="gml:OwnershipAttributeGroup"/>
+	</xs:complexType>
+	<xs:element name="VolcanicAshSIGMET" type="iwxxm:VolcanicAshSIGMETType" substitutionGroup="iwxxm:SIGMET">
+		<xs:annotation>
+			<xs:documentation>A SIGMET that reports the presence of volcanic ash conditions hazardous to flight.  This extends the base SIGMET type by including additional information necessary for volcanoes and volcanic ash.</xs:documentation>
+		</xs:annotation>
+	</xs:element>
+	<xs:complexType name="VolcanicAshSIGMETType">
+		<xs:complexContent>
+			<xs:extension base="iwxxm:SIGMETType">
+				<xs:sequence>
+					<xs:element name="supplementaryAnalysisCollection" minOccurs="0" maxOccurs="unbounded">
```

Comment:
supplementaryAnalysisCollection should be available for all SIGMET types and used only by VA SIGMET, may be restricted by a Schematron rule. Or only TC SIGMET should be extended by supplementaryAnalysisCollection .

I think a more universal data model is better. It also follows Choy's solution for tropicalCyclonePosition, which is presented in all kind of SIGMETs, but it is used only by TC SIGMET.

Reply:
My original intention is to represent the objects before AND with analysisCollection and those after by supplementaryAnalysisCollection.  Your interpretation is obviously different (hence using analysisCollection for VA descriptions before and after AND, and using analysisCollection to represent TC position + CB and supplementaryAnalysisCollection for CB only).  I concur this is a better approach and have implement this in commit https://github.com/wmo-im/iwxxm/commit/14b70a04a0e06106377cd927d6dfecf9ca62f08f

Reply:
After thinking for a while, it seems to me that the location of TC position within the XML document in previous update did not address @jkorosi 's concern that the intensity change is referring to the TC itself and not the CB cloud.  Therefore I have moved the TC position to within <iwxxm:member> and now it should looks exactly like the TAC message; that both TC centre and CB cloud are within the same <iwxxm:member> block described by an @intensityChange.  The revised implementation can be found in commit https://github.com/wmo-im/iwxxm/commit/c29b4a47621b7a942f993e232f44449ea74f88ce

Reply:
I am comfortable with the current proposal.